src/safeposix/syscalls/net_calls.rs

```diff
@@ -131,112 +129,101 @@ impl Cage {
     }
 
     fn bind_inner_socket(&self, sockhandle: &mut SocketHandle, localaddr: &interface::GenSockaddr, prereserved: bool) -> i32 {
-        if localaddr.get_family() != sockhandle.realdomain as u16 {
+        if localaddr.get_family() != sockhandle.domain as u16 {
             return syscall_error(Errno::EINVAL, "bind", "An address with an invalid family for the given domain was specified");
         }
-
+    
         if sockhandle.localaddr.is_some() {
             return syscall_error(Errno::EINVAL, "bind", "The socket is already bound to an address");
         }
-
-        let intent_to_rebind = sockhandle.options & (1 << SO_REUSEPORT) != 0;
+    
         let mut newsockaddr = localaddr.clone();
+    
+        let res = match sockhandle.domain {
+            AF_UNIX => self.bind_inner_socket_unix(sockhandle, &mut newsockaddr),
+            AF_INET | AF_INET6 => self.bind_inner_socket_inet(sockhandle, &mut newsockaddr, prereserved),
+            _ => {return syscall_error(Errno::EINVAL, "bind", "Unsupported domain provided");}
+        };
 
-        if sockhandle.realdomain == AF_UNIX {
-            // create fake IPV4 addr
-            let ipaddr = interface::V4Addr {s_addr: u32::from_ne_bytes([127, 0, 0, 1])};
-            let innersockaddr = interface::SockaddrV4{sin_family: AF_INET as u16, sin_addr: ipaddr, sin_port: 0, padding: 0};
-            newsockaddr = interface::GenSockaddr::V4(innersockaddr);
+        if res != 0 {
```

Comment:
I think we can remove this res check no with the refactor @yizhuoliang

Reply:
yes, removed